textile/chat-features.textile

```diff
@@ -9,5 +9,495 @@ jump_to:
 
 h2(#overview). Overview
 
-Section dedicated to new chat feature development. TBC
+This document outlines the feature specification for the Ably Chat product.
 
+The key words "must", "must not", "required", "shall", "shall not", "should", "should not", "recommended",  "may", and "optional" (whether lowercased or uppercased) in this document are to be interpreted as described in "RFC 2119":https://tools.ietf.org/html/rfc2119.
+
+The over-the-wire protocol for Chat, as well as implementation details, may be found on the chat protocol page.
+
+h2(#overview). Chat Specification Version
+
+* @(CHA-V1)@ **Specification Version**: This document defines the Ably chat library features specification ('features spec').
+** @(CHA-V1a)@ The current version of the Chat library feature specification is version @0.1.0@.
+
+h2(#general). General Principles
+
+* @(CHA-GP1)@ As far as is practicable, the implementation details such as underlying Realtime Channels should be hidden from the public API. This allows developers to interact with Chat without having to understand many low-level primitives.
+* @(CHA-GP2)@ The public API should avoid implicit operations as a side-effect to some other operation. For example, adding a subscriber to messages in a Chat Room should not automatically trigger that Room to attach. This is in contrast to the current core SDKs. Avoiding side-effects provides a clean, easy to understand API.
+* @(CHA-GP3)@ Wherever possible, Chat features should be exposed in the public API as properties of their parent. For example, @messages@ would be considered a property of a @room@ object. This allows for greater composability and extensibility in the future.
+* @(CHA-GP4)@ Avoid overloading methods and optional parameters. Prefer object-type parameters wherever practical and idiomatic.
+
+h2(#rooms). Rooms
+
+h3(#rooms-general). General Information
+
+A Room is the atomic unit of Chat. All chat operations are performed in the context of a room that a client is currently attached to. As the Room is the atomic unit, its state should be considered as the combination of the states of underlying resources.
+
+h3(#rooms-status). Room Statuses
+
+The status of any given Chat Room is the combination of the states of all of its constituent contributors. For more information on this, see "@Room Lifecycle@":#rooms-lifecycle.
```

Comment:
Can we please use some other keyword than `contributors`, this is a super confusing keyword and takes spec into whole different direction.

Reply:
You can rather use `room elements`, `room objects` etc

Reply:
I am open to any other keyword but surely not `contributor`

Reply:
The term `contributor` was chosen because in the context of the Room Lifecycle, a contributor is something that contributes to the room lifecycle and status. I clarify this point in the pointed room lifecycle section - which I can try to make clearer.

`Element` and `Object` are non-descriptive of what the things actually do or mean.

Reply:
sure, `element`, and `object` are non-descriptive when used individually, but they do mean something when we say `room object` or `room element`, although less descriptive, they convey the meaning.

Reply:
When `contributor` is used individually, it's difficult to guess if it's related to `chat` object or `room` object.
Our current implementation is mainly resolved around `room`, so it will be great if we can make it explicit there.